docs/classes/index.rst

```diff
@@ -2,37 +2,137 @@
 Classes
 =======
 
+Classes in kotlin are pretty straighforward, I would say more than python's as well!
 
-1) Defining a class with a method but no constructor, P has “self”
+1) Defining a class with a method but no constructor
 
-2) Creating instance of class does NOT require new in either
+A simple class (w/o a constructor) can be written like this:
 
-3) Class variables access in P via self or class name
+.. code-block:: kotlin
 
-4) P can assign instance attributes whenever it wants (within __ limitations), change types whenever, no concept of public/private
+    class Example {
+        
+    }
 
-5) Constructors
+To add methods to a class you simply have to write a function within that class!
 
-6) Binding of constructor arguments to instance attributes (assignment, usage)
+Example of a class with a method:
+
+.. code-block:: kotlin
+
+    class Example {
+        fun sayHi() {
+            println("Hi there!")
+        }
+    }
+
+By default any function/property in a class is public, that means that code outside the class can access the functions/properties of that class.
+If we want to make a function be only accessible within the class we have to declare it as private.
+
+.. code-block:: kotlin
+
+    class Example {
+        private fun sayHi() {
+            println("Hi there!")
+        }
+    }
+
+Now ``Example.sayHi`` is private, that means that that function is not accessible outside of the Example class.
+
+
+2) Creating instance of a class
+
+to access that method we must initialise that class
+
+that can be done like so:
+
+.. code-block:: kotlin
+
+    val a = Example()
+    a.sayHi()
+
+3) Class variables access in Example via class name
+
+In python to access a class property from within that class we use “self”.
+
+In kotlin (from my understanding) there are two ways to access the class properties from within the class.
+
+First one is to..well..just use the property names normally...that was too simple wasn't it?
+
+Second one is to declare a companion object (its properties are private only) and either use the property names as usual, or use the class name in the same way we use “self” in python.
```

Comment:
We can get rid of the `eg` by just ending this paragraph with `:`

Reply:
👍